### PR TITLE
Refactoring create post

### PR DESCRIPTION
**Pull request by EvgeniyaYakunina**
### **ImagePostModal Component:**

A new modal component for displaying and managing image posts.
Supports creation, editing, and deletion of posts.
Features a step-based approach for creating posts (cropping, filters, publication).
Includes a confirmation modal for closing the post creation process without saving changes.
Utilizes the useImagePostModal custom hook for managing state and behavior.

**useImagePostModal Custom Hook:**

A custom hook for managing the state and behavior of the ImagePostModal component.
Handles post creation, editing, and deletion processes.
Provides functions for changing steps, confirming deletes, and managing modals.
State: step, isEditing, isOpenModal, and various modal visibility states.

**PostForm Component:**

A reusable form component for handling post descriptions.
Supports both creating and editing posts.
Features controlled textarea with zod-validation.

**Post Component:**

A section of the modal representing different steps in the post creation process.
Displays the form, profile info, and location input based on the current step.
Props: avatar, handleClose, postId, selectedImage, step, and userId.
This PR aims to provide a comprehensive and user-friendly solution for managing image posts within the application. The components and custom hook are designed to be reusable and modular, promoting better code organization and maintainability.
1t

### CODE REVIEW

File: src/features/posts/ui/post/post.tsx [hunk 1]
```diff
@@ -1,58 +1,60 @@
...
+import { Typography } from '@photo-fiesta/ui-lib'
 
 import styles from './post.module.scss'
 
 type PostProps = {
   avatar: Avatar[] | undefined
   handleClose: () => void
+  postId: number
   selectedImage: null | string
   step: 'cropping' | 'filters' | 'publication'
```
Comment (Lelik7777) [inline src/features/posts/ui/post/post.tsx]: у тебя это повторяется, может стоит создать тип и его юзать

File: src/features/posts/ui/post/post.tsx [hunk 2]
```diff
@@ -1,58 +1,60 @@
...
   userId: number | undefined
 }
 
-export const Post = forwardRef<HTMLFormElement, PostProps>(
-  ({ avatar, handleClose, selectedImage, step, userId }, ref) => {
-    const { control, errors, onSubmit } = useImagePostModal({ handleClose, selectedImage })
+/**
+ * The Post component represents a section of modal with different steps like cropping, filters, and publication.
```
Comment (Lelik7777) [inline src/features/posts/ui/post/post.tsx:20]: add using example

File: src/features/posts/ui/post/post.tsx [hunk 3]
```diff
@@ -1,58 +1,60 @@
...
-              <Typography>
-                <label>Add location</label>
-                <textarea className={styles.location} name={'location'} placeholder={'NewYork'} />
-              </Typography>
+export const Post = ({ avatar, handleClose, postId, selectedImage, step, userId }: PostProps) => {
+  if (step === 'publication') {
+    return (
+      <section className={styles.formSection}>
+        <div className={styles.profileInfo}>
+          <ProfileAvatar avatarOwner={avatar?.[0].url} />
```
Comment (Lelik7777) [inline src/features/posts/ui/post/post.tsx:28]: extract to constant

File: src/features/posts/ui/post/postForm/postForm.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+/** Validation schema for the post description */
+const postDescriptionSchema = z.object({
+  description: z
+    .string()
+    .min(1, 'Minimum number of characters 1')
+    .max(500, 'Maximum number of characters 500'),
+  // location: z.string(),
+  //TODO: check type of this field(location)
+})
```
Comment (Lelik7777) [inline src/features/posts/ui/post/postForm/postForm.tsx]: move to utils/commonFormUtils/ts

File: src/features/posts/ui/post/postForm/postForm.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    defaultValues: { description: post?.description || '' },
+    mode: 'onBlur',
+    resolver: zodResolver(postDescriptionSchema),
+  })
+
+  /** Submit function for create description in post modal */
+  const onSubmit = handleSubmit(async (data: FormValues) => {
+    try {
+      if (!selectedImage) {
+        throw new Error('No image selected')
```
Comment (Lelik7777) [inline src/features/posts/ui/post/postForm/postForm.tsx]: use toaster

File: src/features/posts/ui/post/postForm/postForm.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+      handleClose()
+    } catch (error) {
+      handleErrorResponse<FormValues>({ badRequestSchema, error, setError })
+    }
+  })
+
+  /**
+   * This constant is used to render the `FormTextArea` component with the appropriate label and placeholder
+   * based on whether the post is being created or edited.
+   */
```
Comment (Lelik7777) [inline src/features/posts/ui/post/postForm/postForm.tsx]: all logic from this component move to custom hook usePostForm

**Comment by Lelik7777** [comment]
add screenshot for PR

File: src/features/posts/ui/post/postForm/postForm.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+type PostFormProps = {
+  handleClose: () => void
+  isEditing: boolean
+  postId: number
+  selectedImage?: null | string
+}
+/**
+ * The PostForm component handles creating and editing posts description with controlled textarea and zod-validation.
+ */
```
Comment (Lelik7777) [inline src/features/posts/ui/post/postForm/postForm.tsx]: move to usePostForm

File: src/features/posts/ui/post/postForm/postForm.tsx (hunk 5 above)
Comment (Lelik7777) [inline src/features/posts/ui/post/postForm/postForm.tsx]: move to usePostForm

File: src/features/posts/ui/post/postForm/postForm.tsx [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+export type FormValues = z.infer<typeof postDescriptionSchema>
+
+type PostFormProps = {
+  handleClose: () => void
+  isEditing: boolean
+  postId: number
+  selectedImage?: null | string
+}
+/**
+ * The PostForm component handles creating and editing posts description with controlled textarea and zod-validation.
```
Comment (Lelik7777) [inline src/features/posts/ui/post/postForm/postForm.tsx:15]: add using example

File: src/features/posts/ui/post/postForm/postForm.tsx (hunk 5 above)
Comment (EvgeniyaYakunina) [reply src/features/posts/ui/post/postForm/postForm.tsx]: поменяла на toast

File: src/features/posts/ui/post/post.tsx (hunk 1 above)
Comment (EvgeniyaYakunina) [reply src/features/posts/ui/post/post.tsx]: вынесла в тип

File: src/features/posts/ui/post/postForm/postForm.tsx (hunk 6 above)
Comment (EvgeniyaYakunina) [reply src/features/posts/ui/post/postForm/postForm.tsx]: вынесла логику в хук

**Comment by EvgeniyaYakunina** [comment]
run build production

**Pull request merged**